core/src/main/java/com/graphhopper/routing/subnetwork/PrepareRoutingSubnetworks.java

```diff
@@ -38,29 +36,31 @@
 /**
  * Removes nodes/edges which are not part of the 'main' network(s). I.e. mostly nodes with no edges at all but
  * also small subnetworks which could be bugs in OSM data or 'islands' or indicate otherwise disconnected areas
- * e.g. via barriers or one way problems - see #86.
- * <p>
+ * e.g. via barriers or one way problems - see #86. Subnetworks are removed by disabling access to the corresponding
+ * edges for a given access encoded value. It is important to search for strongly connected components here (i.e.
+ * consider that the graph is directed). For example, small areas like parking lots are sometimes connected to the whole
+ * network through a single one-way road (a mapping error) and have to be removed because otherwise the routing fails
+ * when starting from such a parking lot.
  *
  * @author Peter Karich
  * @author easbar
  */
 public class PrepareRoutingSubnetworks {
     private final Logger logger = LoggerFactory.getLogger(getClass());
     private final GraphHopperStorage ghStorage;
-    private final List<FlagEncoder> encoders;
-    private final List<BooleanEncodedValue> accessEncList;
+    private final List<PrepareJob> prepareJobs;
     private int minNetworkSize = 200;
-    private int subnetworks = -1;
 
-    public PrepareRoutingSubnetworks(GraphHopperStorage ghStorage, List<FlagEncoder> encoders) {
+    public PrepareRoutingSubnetworks(GraphHopperStorage ghStorage, List<PrepareJob> prepareJobs) {
         this.ghStorage = ghStorage;
-        this.encoders = encoders;
-        this.accessEncList = new ArrayList<>();
-        for (FlagEncoder flagEncoder : encoders) {
-            accessEncList.add(flagEncoder.getAccessEnc());
-        }
+        this.prepareJobs = prepareJobs;
     }
 
+    /**
+     * The subnetwork removal removes components with less than {@link #minNetworkSize} nodes from the graph if it is
+     * run node-based. For edge-based subnetwork removal it removes components with less than 2*{@link #minNetworkSize}
+     * (directed) edges.
+     */
```

Comment:
We could make this more consistent and also count the number of nodes for edge-based subnetwork removal, but I am not sure if its worth it. For edge-based subnetwork search we only get the edge keys of the components and would have to collect the set of nodes adjacent to these edges to find out how many nodes a component has.